[#14682] Minimal CI build/test

rigazilla opened the pull request:
fixes #14682 

minimal CI that only runs tests for 14.0.x clients 

Replaces #14683

tristantarrant commented:
I have removed maven-settings.xml in 14.0.x as I've cleaned some things up for the 14.0.35.Final release. Can you fix the scripts ?

rigazilla commented:
> I have removed maven-settings.xml in 14.0.x as I've cleaned some things up for the 14.0.35.Final release. Can you fix the scripts ?

yep, removed maven-settings.xml from all the actions
